src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -616,13 +616,28 @@ unique_ptr<BoundQueryNode> Binder::BindSelectNode(SelectNode &statement, unique_
 	}
 
 	// push the GROUP BY ALL expressions into the group set
+
 	for (auto &group_by_all_index : group_by_all_indexes) {
 		auto &expr = result->select_list[group_by_all_index];
 		auto group_ref = make_uniq<BoundColumnRefExpression>(
 		    expr->return_type, ColumnBinding(result->group_index, result->groups.group_expressions.size()));
 		result->groups.group_expressions.push_back(std::move(expr));
 		expr = std::move(group_ref);
 	}
+	set<idx_t> group_by_all_indexes_set;
+	if (!group_by_all_indexes.empty() && !statement.having) {
+		for (idx_t i = 0; i < group_by_all_indexes.size(); i++) {
+			group_by_all_indexes_set.insert(i);
+		}
+		D_ASSERT(result->groups.grouping_sets.empty());
+		result->groups.grouping_sets.push_back(group_by_all_indexes_set);
+		// rebind the having in this case
+		if (statement.having) {
+			HavingBinder having_binder(*this, context, *result, info, statement.aggregate_handling);
```

Comment:
I'm a bit unsure about this part - binding can be destructive, so rebinding the same expression again is not necessarily safe.

Could you expand a bit on why this is necessary?